src/ajax.js

```diff
@@ -562,14 +562,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Apply prefilters
+		inspectPrefiltersOrTransports( prefilters, s, options, jqXHR );
+
 		// Convert data if not already a string
 		if ( s.data && s.processData && typeof s.data !== "string" ) {
 			s.data = jQuery.param( s.data, s.traditional );
 		}
 
-		// Apply prefilters
-		inspectPrefiltersOrTransports( prefilters, s, options, jqXHR );
```

Comment:
I think moving prefilters ahead of string coercion is a breaking change with respect to custom prefilters, cf. https://api.jquery.com/jQuery.ajaxPrefilter/ (emphasis mine):
> <code>originalOptions</code> are the options as provided to the <code>$.ajax()</code> method, **unmodified** and, thus, without defaults from <code>ajaxSettings</code>

Is there a way to instead establish something like prefilter phases?

Reply:
The line that now happens after the prefilters is only setting `data`, not options. Is there any change that I'm missing?

BTW, I would also consider this a breaking change regardless of the above, that's why I'm targeting it at v4.